test/rlcIntegration/bodyComplextRest.spec.ts

```diff
@@ -24,33 +25,32 @@ describe("BodyComplex Rest Client", () => {
         const result = await client.path("/complex/basic/valid").get();
 
         try {
-          if (result.status !== "200") {
+          if (isUnexpected(result)) {
```

Comment:
I may lack some context but I am not quite sure adding this helper function 'isUnexpected()' would help as user still need to parse the error code themselves. 
And can we use 'result.IsSuccessful' instead?
I suggest we shall do some investigation about better error handling for RLC. 
So can we postpone this after RLC 1.0 GA?    cc @qiaozha

Reply:
I was talking to @bterlson and I'd like to propose a slightly different approach. Which is:

* We don't generate a missleading status code. For the default case we give status a type of `string` which represents correctly an unknown set of strings
   * This will initially make TS collapse the type of status for both types the Success and Error to string, and intellisense provide no suggestions. Here's where `isUnexpected` comes to help.
* As soon as we validate with `isUnexpected` we can narrow down the types and get suggestions again.


Here are 2 playgrounds for the current vs proposed


[Current](https://www.typescriptlang.org/play?#code/PTAEBUAsEsGdTqAhqATgU1gBwPYDtZ010sNC8AXaPAc2VFgFcBjZzWAOgCgRQB5PABsAnqABMABgkAacRICMyPABM5AFmQZQzfLGjL0GVU1btMPMEjyicFSIYYUkFRvB0HQAd2iDBoAEZEOgT6huiqVqCGqDioXNQUhgBmSGygAOLoFACqhKiwAMosbLCwAEqYuAREAN5coI7OrgBcoABEkhJtoAA+7ZLy3X0dEmpt0vUBOMrCrTUAbkiCjOitsBSo1DQAvly7FhAw8IgoNOh4hktRqDGo2kiEsp5EyvgA5BSgANZ4OJ4B6CSsSIkCsygOnhgzEgjRcbmmRGeb1UZ0+sBwXiIriIbQMKUYggobQOkD+6HmDjxSAJn0Qv0+GCWInoi0E+lAkAoFCwsNc2gR3ASyVSREyOTysAAIoDqYSKthdLVJusmrBWriZTTxpN-NNZqAatFYnN3KtQHhGABbQKoADcoEt7CQZzWGy2212+wOAEkvFZPhQMYFkP5BERA6AaIwkKhVHZw8IsIjoHZQHZEElGHhmFR8EpVBgXKg8Gn7GnE+HQZ9QfAVXCAMII5DwKyidabWigIF3bH5NNBohIUPhnAHaHoZhfLuxZDMRhOPx1vmmzhcTPZ3MluDZC4ADyTOfCAAoyFVCK0xblDIViux5WeiH1LxLpfi5ZVFaAAJStU+fxDPter6yhQ96fnUDQNIWjDFqAACEADaIxdLIIyDKhkhiG0AC6HDUMwywGLAJ4ftUHBLrAX57FwBwALI4JOXZZjm0B5nYzjEEWBD0H+1RcAYBExkQ64sXmqJXvkR4-hkWQSTepilGB1S9DJ4pAZq74KtUto0bwYpUJ28bEFphBcME6zGQ+oAALyRrJEpSTpBwAJo4IwHJIBSvLwh4TA0Gc6ysdxSQxJapZEKikY4HBBzgKgojMKgFB+AA1I4WAiggJZGeglpYBQogAI6MLYmDcOZOBhhwgg4DQJEmeg5FOHCNnWbZbRtFR8RJKAR7bnuB6JMo9UPl+X4GpMDS8K57nMFYHzEEyCX2Ix3YMAe0BJNAzD8kRXgpqS873IFhkYq2HCgM0bxHpNCBJCNipNaqrXtQArAonUTZB338gQlWNTVdVtOkzjoJ4SCiEaqCdbd+w-bwUCIIgrLsmtRk6HlPiGLI-hHd4vjmuSDiglgSYlg8oBUjSCDwPSPHoFcAAS4DgAACt5u3oLdm0PWRFEve0VOEp9EE-Q0FVVYDR5tIj8D434FwUncJNkzD31w5BdgxP8vGEBwuozBwUM6V6EsA7VvN6wbwgcKyKxfk5QA) vs [Proposed](https://www.typescriptlang.org/play?#code/PTAEBUAsEsGdTqAhqATgU1gBwPYDtZ010sNC8AXaPAc2VFgFcBjZzWAOgFgAoEUAPJ4ANgE9QAJgAMUgDSSpARmR4AJgoAsyDKGb5Y0Vegzqmrdpl78kecTgqRjDCkgqN4eo6ADu0YcNAAIyI9AkNjdHUbUGNUHFReagpjADMkNlAAcXQKAFVCVFgAZRY2WFgAJUxcAiIAb15QZ1d3AC5QACJpKQ7QAB9O6UVega6pDQ7ZRqCcVVF2uoA3JGFGdHbYClRqGgBfXn2eKzAoREQUGnQ8YxWY1DjUXSRCeW8iVXwAcgpQAGs8HDeILoFLxIiQGyqY4+GDMSDNNweWZEN6fdSXH6wHA+IjuIgdIxpRjCCgdaGQQHoRZOQlIYk-RAAn4YFZiejLYSGUCQCgULAI9y6ZHcHhJVLpIjZPIFWAAERBdJJVWw+nq002LVgGy2OymPCagVm81AdVi8QWnnWoDwjAAtsFUABuUC29hIS7a7a0XaHQ68aEASR8Nh+FGxwWQgWERDDoBojCQqHUDhjoiwKOgDlADkQKUYeGYVHwKnUGDcqDw2cc2bTMYhPwh8A1iIAwsjkPAbOJNl66KDHnjCtnw0QkFGYzhoXD0MxfqB+8hmIwXAFm4LLZxeHmC0XK3BctcAB7pwuRAAUZBqhHaUvyxmKpXYyqvRAGt5l8qJSuqqtAAEp2kvX9EHfe9P0VChn1-Bp9SaYhy0rABCABtMYenkMZhgw6QJA6ABdDhqGYVYjFgC8f1qDg11gP8Dn9PgwAAWRwWd53zQtoGLBxXHgxgK07YgVVqXgjGIxMiG3DjiwxO9CjPACshyWSH3McooNqfpFOlMCFXpdTCEdej+ClKhaCrIggOEnhQk2QSX1AABeOMlJleTDKOBjQFEHBGFAYQcBwQhzIFJEvAwPRbVdNRXE4ghgX7Ig8ETOJfDMj5vDwaFwFQcRmFQCgAgAamcLAJQQSsUxiW0sAocQAEdGHsTAOFAAA5bEmBoS5Nli2BeBsnBow4fyaHIoTCColxEUchynI6DpaI86AUlAM99yPE9klUMaXz-P8TWmJp+AATR8p48G+YhWVyxxWIXbAZ2W6BmCFUifEzCllyeHqzNjLtDoQFIdtVSbNRmuaAFYlAWg7YLguCBqGkazw6TJXHQbwkHEM1UAWgG-Tho6ThgeBEA5LkF0qiKsD8Yx5ECL7fH8a0qScCEsHTStnlAWl6QQeAmXoFkAgACXAcAAAUQte9AAeW4HKOo8HOl5kkYZg+H4cR9BhpwUaOlOeAmYCa5qUednObxwmCfhhwUrskHDTmDgcfcgn+BFiIvLOiFqWly0GEYLrMF3eBWkW-hsty-KipKsrqGC9Bqtq0AGqazdrP0QadeRyyJqV2a5qt7XddGvOdad0QOA5NY-3c3ggA)



/cc - @bterlson. @lirenhe, @MaryGao, @qiaozha

Reply:
Compared with the `default` type I would prefer the native `string` type more. That would be good to narrow down the error types without casting or any type checking errors with the helper func.

Reply:
Just have one simple question, if isUnexpected, should the error and message be the properties of result not result.body ? 
Otherwise, this proposal looks good to me. @lirenhe  how do you think of it?

Reply:
As I know, Go SDK uses result.properties but I can't say which one is better.
And overall, the proposal looks good to me. Thanks @joheredi for the update.